source/SH3/system/glcontext.cpp

```diff
@@ -45,7 +152,22 @@ context::context(sh3_window& hwnd)
     SDL_GL_SetAttribute(SDL_GL_BUFFER_SIZE, 32);
 
     SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
-}
+
+    // Enable debug output (may work for non-debug contexts, depending on OpenGL implementation)
+    // Kinda risky because it __mayyy__ not work on all systems/graphics drivers, but GL 3.3 gives
+    // us the most compatability.
+    if(SDL_GL_ExtensionSupported("GL_KHR_debug"))
+    {
+        glEnable(GL_DEBUG_OUTPUT);
+        glEnable(GL_DEBUG_OUTPUT_SYNCHRONOUS);
+        glDebugMessageCallback(debugCallback, nullptr);
+        glDebugMessageControl(GL_DONT_CARE, GL_DONT_CARE, GL_DONT_CARE, 0, nullptr, GL_TRUE);
```

Comment:
> We need to dynamically load the address using SDL_GL_GetProcAddress, otherwise the startup of our application will fail on systems where this extension is not supported.

I thought GLEW did that for us?

> We can additionally check for "ARB_debug_output", which functions are named differently but have an identical signature so it should work the same.
If you want to go the extra mile you can also try "AMD_debug_output", though there the functions have a differing signature so you need to change a bit more.

Okay I'll get on adding that just to make it extra water-tight.

Reply:
> I thought GLEW did that for us?

Oh, you're right. You can also shorted the check to `if(GLEW_KHR_debug)` with GLEW :)